src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -132,14 +132,14 @@ private static void ResolveTocModelItems(List<TableOfContentsInputItem> tocModel
                         var (referencedTocContent, referenceTocFilePath, isYamlToc) = ResolveTocHrefContent(hrefType, tocModelItem.Href, filePath, resolveContent);
                         if (referencedTocContent != null)
                         {
-                            tocModelItem.Items = LoadInputModelItems(referencedTocContent, isYamlToc, referenceTocFilePath, rootPath, resolveContent, resolveHref, parents);
+                            tocModelItem.Items = LoadInputModelItems(referencedTocContent, isYamlToc, referenceTocFilePath, resolveContent, resolveHref, parents);
                             tocModelItem.Href = topicHref;
                         }
                     }
                     else
                     {
                         tocModelItem.Href = string.IsNullOrEmpty(topicHref)
-                            ? resolveHref?.Invoke(filePath, href, rootPath) ?? tocModelItem.Href
```

Comment:
>rootPath [](start = 66, length = 8)

where is rootpath?

Reply:
i think the interface should contains all the necessary inputs to generate expected output, current resolvehref can't give me enough information for that.

---
In reply to: [193270998](https://github.com/dotnet/docfx/pull/2839#discussion_r193270998) [](ancestors = 193270998)